src/torchmetrics/functional/regression/kendall.py

```diff
@@ -0,0 +1,396 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.utilities.checks import _check_same_shape
+from torchmetrics.utilities.data import _bincount, dim_zero_cat
+from torchmetrics.utilities.enums import EnumStr
+
+
+class _MetricVariant(EnumStr):
+    """Enumerate for metric variants."""
+
+    A = "a"
+    B = "b"
+    C = "c"
+
+    @classmethod
+    def from_str(cls, value: str) -> Optional["EnumStr"]:
```

Comment:
could this `from_str` be implemented in the base class?
it seems to be in every subclass with only slight variations

Reply:
I will check this what I can do it here